app/src/main/java/org/mozilla/focus/utils/StorageUtils.java

```diff
@@ -91,8 +91,17 @@ static File getFirstRemovableMedia(@NonNull Context ctx) {
         final File[] files = ctx.getExternalMediaDirs();
         for (final File file : files) {
             // on some devices such as Oppo, it might return null
-            if ((file != null) && Environment.isExternalStorageRemovable(file)) {
-                return file;
+            if ((file != null)) {
+                // Environment will invoke internal API to check volume,
+                // and throws an IllegalArgumentException if the target path has any problem
+                try {
+                    if (Environment.isExternalStorageRemovable(file)) {
```

Comment:
https://developer.android.com/reference/android/os/Environment.html#isExternalStorageRemovable() states it throws IllegalArgumentException if the path is not a valid storage device.

Let's not catch everything but just IllegalArgumentException since some thing/api may have changed and we want to know when it is broken.